src/core/server/ui_settings/ui_settings_client.ts

```diff
@@ -94,6 +94,12 @@ const UiSettingScopeReadOptions = [
     autoCreateOrUpgradeIfMissing: false,
     ignore404Errors: true,
   },
+  {
+    scope: UiSettingScope.DASHBOARD_ADMIN,
+    ignore401Errors: false,
+    autoCreateOrUpgradeIfMissing: false,
```

Comment:
I see you have the creation logic in the wrapper, why not leverage `autoCreateOrUpgradeIfMissing`?

Reply:
I think the reason is we want to have custom logic (i.e. create with permission) so we have to write a custom create function.

Reply:
That makes sense.